docs/1.getting-started/6.server-directory.md

```diff
@@ -0,0 +1,59 @@
+---
+navigation.icon: uil:laptop-connection
+description: Using server directory to create middlewares, plugins and API easily.
+---
+
+# Server Directory
+
+Server directory can be used for creating API routes to implement reusable data fetching from remote server easily, middlewares for intercepting requests and responses and plugins to hook into lifecycle hooks.
+
+Each file under these folders must return a default export `defineEventHandler()`.
+
+## Server Routes
+
+Anything under `~/server/api` folder will create API routes to fetch data within API route.
+
+```ts [server/api/hello.ts]
+export default defineEventHandler(() => {
+    return {
+      greetings: 'traveler'
+    }
+})
+```
+
+```vue [components/Greet.vue]
+<script setup lang="ts">
+const { data } = await useFetch('/api/hello')
+</script>
+
+<template>
+  <div>
+     {{ data }}
+   </div>
+</template>
+```
+
+This server route will return a data object that contains an object with content of `{greetings: 'traveler'}` printed in the template tag.
+
+Server route file will automatically match with `GET` method and it can be changed by adding method names to file name such as: `.post`, `.put`, `.delete`.
+Routes may accept parameters, body and query parameters to get some values.
+
+:ReadMore{link="/docs/guide/directory-structure/server#matching-route-parameters"}
+
+## Server Middleware
+
+Nuxt will read anything under `~/server/middleware` folder and create middlewares for you to tweak request information.
+
+```ts [server/middleware/log.ts]
+export default defineEventHandler((event) => {
+  event.context.auth = { user: 123 }
+})
+```
+
+:ReadMore{link="/docs/guide/directory-structure/server#server-middleware"}
+
+## Server Plugins
```

Comment:
Maybe we could also add a section about the `utils` directory where content inside is auto-imported.

Reply:
Good idea!

Reply:
Added basic explanation for `~/server/utils`. Haven't used it yet properly so it might be false/lacking.

Reply:
Currently adding it to the Nitro documentation https://github.com/unjs/nitro/pull/1451. Will review your change.